resources/components/BOMForm/BOMForm.tsx

```diff
@@ -0,0 +1,179 @@
+import React,{useState} from 'react'
+import {InputNumber, Button, Empty, Table, Row, Col, Select, Typography} from 'antd'
+import {InventoryItemStateType} from 'models/inventory'
+import {BomMaterialStateType} from 'models/bom-material'
+import {StoreType, DispatchArgumentType} from '@types'
+import {connect} from 'react-redux'
+import {DeleteButton} from '@components'
+import {DeleteOutlined} from '@ant-design/icons'
+import {getBomMaterial} from 'services/bom-material'
+
+
+type BOMFormPropType = {
+    dispatch: (arg: DispatchArgumentType) => void,
+    InventoryItem: InventoryItemStateType,
+    BomMaterial: BomMaterialStateType,
+    inventoryId: number,
+}
+
+const BOMForm = (
+     {
+     dispatch,
+     InventoryItem,
+     BomMaterial,
+     inventoryId
+ }:BOMFormPropType
+)  => {
+    const {table, form} = BomMaterial
+    const {Title} = Typography
+    const [isEdit, setEdit] = useState<boolean>(false)
+    const changeFormData = (key: string, value: any) => dispatch({type: 'BOM_MATERIAL_PUSH_FORM_DATA', payload: {key, value}})
+    const changeFormQuantity = (key: number, value: any) => dispatch({type: 'BOM_MATERIAL_CHANGE_FORM_QUANTITY', payload: {key, value}})
+    const deleteBom = (key: number) => dispatch({type: 'DELETE_BOM', payload: key})
+    const resetState = () => dispatch({type: 'RESET_BOM_STATE'})
+
+    const fetchBomList = (id: number) => {
+      getBomMaterial(id).then((response: any) => {
+        const {data} = response
+        if (data.success) {
+          dispatch({type: 'RESET_BOM_STATE'})
```

Comment:
unsure why resetting BOM state here, should probably call this instead when closing the form or component on dismount

Reply:
also why do we have fetchBomList in 2 places, you can just pass it down from parent component if this is needed